spec/lib/vanagon/project_spec.rb

```diff
@@ -126,6 +127,179 @@
     end
   end
 
+  describe "#load_yaml_settings" do
```

Comment:
Looking at the code for `load_yaml_settings`, it seems like it would be enough to init. a single Project object here and then test each case accordingly. Something like:

```
describe '#load_yaml_settings' do
  subject(:project) { Project.new("project", "mock_platform" }
  describe 'with a git source' do
    it 'fails to load anything' do
      expect(project.load_yaml_settings(git_uri).to raise_error(Vanagon::Error)
    end
  end
  // etc for the other cases, mocking only when you need to mock a specific method
end
```

I don't understand the use of all these blocks here when you're only testing that this method works for a given project instance. The test for whether
```
project 'upstream-test' do |proj|
end
```

returns the right project object after all of that set-up should be handled by the corresponding DSL test.

Reply:
Hmm. I was following the existing conventions in this file, but it would be nice to get rid of a lot of this mocking. I'll do some experiments tomorrow morning.

Reply:
Yeah I figured. It just seems like the existing conventions are a little overkill here.

Reply:
Thanks @ekinanp, I've updated this to remove all of the project definition blocks

Reply:
Thanks @caseywilliams, the tests are a lot clearer now. One question -- is there a reason why mocking https://github.com/caseywilliams/vanagon/blob/VANAGON-130/yaml-settings/lib/vanagon/project.rb#L744-L747 requires so many steps? It seems like it should be enough to expect the Component::Source.source to receive the right params and then return a "mock" source that, when its fetch method is invoked, sets its file field to the YAML file we want (simulating the download). You should be able to mock all of that in a top-level before block.